File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    const lastIndexOf = {}
    
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i]  
        
        if (lastIndexOf.hasOwnProperty(num) && Math.abs(i - lastIndexOf[num]) <= k) {
            return true
        }
            
        lastIndexOf[num] = i
    }
        
    return false
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        previous_index = {}
        
        for idx, num in enumerate(nums):
            if (num in previous_index) and (abs(previous_index[num] - idx) <= k):
                return True
            
            previous_index[num] = idx
            
        return False